src/main/java/alex/mojaki/s3upload/StreamTransferManager.java

```diff
@@ -502,4 +552,16 @@ public void customiseCompleteRequest(CompleteMultipartUploadRequest request) {
     public void customisePutEmptyObjectRequest(PutObjectRequest request) {
     }
 
+    private static class PartNumberComparator implements Comparator<PartETag> {
```

Comment:
can't you just pass somehow a lambda `PartETag::getPartNumber` instead of writing this comparator?

Reply:
Project uses java6. I don't want to change it.